StRoot/StETofMatchMaker/StETofMatchMaker.cxx

```diff
@@ -3926,4 +3909,120 @@ StETofMatchMaker::sortandcluster(eTofHitVec& matchCandVec , eTofHitVec& detector
 	}						
       }// loop over MMMap
   }//loop over counters
+  
+  sortOutOlDoubles(finalMatchVec);
+}
+
+void
+StETofMatchMaker::sortOutOlDoubles(eTofHitVec& finalMatchVec){
+  
+  eTofHitVec overlapHitVec;
+  
+  eTofHitVec tempVecOL        = finalMatchVec;
+  
+  std::vector<int> trackIdVec;
+  
+  for(unsigned int i =0; i< finalMatchVec.size(); i++){
+    
+    if( !(std::find(trackIdVec.begin(), trackIdVec.end(), finalMatchVec.at(i).trackId) != trackIdVec.end())){
+      
+      trackIdVec.push_back(finalMatchVec.at(i).trackId);
+
+      int counterId1 = (finalMatchVec.at(i).sector*100) + (finalMatchVec.at(i).plane*10) + (finalMatchVec.at(i).counter);
+      
+      for(unsigned int j =0; j< finalMatchVec.size(); j++){
+	
+	int counterId2 = (finalMatchVec.at(j).sector*100) + (finalMatchVec.at(j).plane*10) + (finalMatchVec.at(j).counter);
+	
+	if(counterId1 != counterId2  && finalMatchVec.at(i).trackId == finalMatchVec.at(j).trackId){
+	  
+	  if(!(finalMatchVec.at(j).matchFlag % 2))	finalMatchVec.at(j).matchFlag++;
+	  if(!(finalMatchVec.at(i).matchFlag % 2))	finalMatchVec.at(i).matchFlag++;
+	  
+	}
+      }  
+    }else{    
+      continue;
+    }
+  }
+
+  eTofHitVec tmpVec;
+  eTofHitVec OlVec;
+  std::map< Int_t, eTofHitVec >  overlapHitMap;
+
+  tmpVec = finalMatchVec;
+  finalMatchVec.clear();
+  finalMatchVec.resize(0);
+  
+  for(unsigned int i=0; i< tmpVec.size(); i++){ 
+
+    if(tmpVec.at(i).matchFlag%2 == 0){
+      finalMatchVec.push_back(tmpVec.at(i));
+    }else{
+      OlVec.push_back(tmpVec.at(i));
+    }
+  }
+
+  // sort out OlVec
+  for(unsigned int i =0; i < OlVec.size(); i++){
+    overlapHitMap[OlVec.at(i).trackId].push_back(OlVec.at(i));
+  }
+
+  map<Int_t, eTofHitVec >::iterator it;
+  
+  for (it = overlapHitMap.begin(); it != overlapHitMap.end(); it++){
+    
+    eTofHitVec trackVec = it->second;
+    int ind_best = 0;
+    int dr_best = 9999;
+    
+    for(unsigned int n=0; n< trackVec.size();n++){
+      
+      float dr = sqrt((trackVec.at(n).deltaX * trackVec.at(n).deltaX ) + (trackVec.at(n).deltaY * trackVec.at(n).deltaY ));
+      
+      if(dr < dr_best){	    
+	dr_best=dr;
+	ind_best=n;
+	  }
+    }
+    finalMatchVec.push_back(trackVec.at(ind_best));
+  }
+
+  //fix matchFlags 
```

Comment:
The handling of `matchFlag` is somewhat cryptic, especially with magic numbers like 9, 100, 50, etc. These could benefit from either constants or enumerations that explain what these values represent for clarity and maintainability.

Reply:
I agree that this is not optimal, changing the flag would require me to redo parts of the matching procedure and my QA though. In addition the potentially different  treatment of different cases is not mapped out in depth yet and might provide additional information on how a more "intuitive/easy to handle" match flag should look like.
Until I can provide better motivated advice how to make best use of the flag I'd like to keep it as it is.